pubsub/subscriptions/sync_pull.go

```diff
@@ -44,18 +43,17 @@ func pullMsgsSync(w io.Writer, projectID, subName string, topic *pubsub.Topic) e
 	sub.ReceiveSettings.Synchronous = true
 	sub.ReceiveSettings.MaxOutstandingMessages = 10
 
-	cctx, cancel := context.WithCancel(ctx)
-	err = sub.Receive(cctx, func(ctx context.Context, msg *pubsub.Message) {
-		mu.Lock()
-		defer mu.Unlock()
+	// Receive messages for 5 seconds.
+	ctx, cancel := context.WithTimeout(ctx, 5*time.Second)
+	defer cancel()
+
+	// Receive blocks until the passed in context is done.
+	err = sub.Receive(ctx, func(ctx context.Context, msg *pubsub.Message) {
 		fmt.Fprintf(w, "Got message :%q\n", string(msg.Data))
 		_ = msg // TODO: handle message.
 		msg.Ack()
-		if received++; received == 10 {
-			cancel()
-		}
 	})
-	if err != nil && err != context.Canceled {
+	if err != nil && err == context.Canceled {
```

Comment:
It's an error if the context gets cancelled? Why check `!= nil` and `== context.Canceled`?

Reply:
Fixed. This was actually entirely wrong, since instead of checking for `context.Cancelled`, I needed to check for the grpc `CANCELLED` code.